File: read_data.m
function d = read_data()
    N = 100;

    fmt = ['%s', repmat('%f%f%f%f%d%d', 1, N)];
    fid = fopen('in_sample_data.txt');
    data = textscan(fid, fmt, 'delimiter', ',');
    datetime = data{1};

    T = size(data{1}, 1);

    d = {};
    so = zeros(N, T);
    sh = zeros(N, T);
    sl = zeros(N, T);
    sc = zeros(N, T);
    tvl = zeros(N, T);
    ind = zeros(N, T);

    for ii = 0 : N - 1
        so(ii + 1, :) = data{ii * 6 + 2};
        sh(ii + 1, :) = data{ii * 6 + 3};
        sl(ii + 1, :) = data{ii * 6 + 4};
        sc(ii + 1, :) = data{ii * 6 + 5};
        tvl(ii + 1, :) = data{ii * 6 + 6};
        ind(ii + 1, :) = data{ii * 6 + 7};
    end

    fclose(fid);

    d.datetime = datetime;
    d.N = N;
    d.T = T;
    d.so = so;
    d.sh = sh;
    d.sl = sl;
    d.sc = sc;
    d.tvl = tvl;
    d.ind = ind;
end

File: read_data.py
import numpy as np

def read():
    N = 100
    T = 1003
    date = []
    so = np.zeros([N, T]);
    sh = np.zeros([N, T]);
    sl = np.zeros([N, T]);
    sc = np.zeros([N, T]);
    tvl = np.zeros([N, T]);
    ind = np.zeros([N, T]);

    with open('in_sample_data.txt') as f:
        for tt, line in enumerate(f):
            cols = line.split(',')
            date.append(cols[0])
            for ii in xrange(N):
                so[ii, tt] = float(cols[ii * 6 + 1])
                sh[ii, tt] = float(cols[ii * 6 + 2])
                sl[ii, tt] = float(cols[ii * 6 + 3])
                sc[ii, tt] = float(cols[ii * 6 + 4])
                tvl[ii, tt] = float(cols[ii * 6 + 5])
                ind[ii, tt] = float(cols[ii * 6 + 6])

    d = {}
    d['date'] = date
    d['so'] = so
    d['sh'] = sh
    d['sl'] = sl
    d['sc'] = sc
    d['tvl'] = tvl
    d['ind'] = ind
    d['N'] = N
    d['T'] = T

    return d

if __name__ == '__main__':
    read()
